dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/utils/DependentExecute.java

```diff
@@ -129,51 +140,87 @@ private DependResult calculateResultForTasks(DependentItem dependentItem,
 
     /**
      * depend type = depend_all
-     *
-     * @return
      */
-    private DependResult dependResultByProcessInstance(ProcessInstance processInstance) {
-        if (!processInstance.getState().isFinished()) {
-            return DependResult.WAITING;
-        }
+    private DependResult dependResultByProcessInstance(ProcessInstance processInstance, DateInterval dateInterval) {
         if (processInstance.getState().isSuccess()) {
+            List<ProcessTaskRelation> taskRelations = processService.findRelationByCode(processInstance.getProcessDefinitionCode(),
```

Comment:
The workflow dependent only need check the process instance state.
There are scenarios in which only some tasks are run, resulting in a successful workflow.